internal/credential/service_list_stores.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/pagination"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// SubtypeStoreService defines the interface expected
+// to gather information about credential stores.
+type SubtypeStoreService interface {
+	EstimatedStoreCount(context.Context) (int, error)
+	ListDeletedStoreIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListCredentialStores(context.Context, []string, ...Option) ([]Store, error)
+}
```

Comment:
The SubtypeStoreService is implemented by both the static and vault credential repos.